metabase/metabase
# 44723 human readable coercion strategies

Closes https://github.com/metabase/metabase/issues/44723

### Description
Over a year ago, the field editor was refactored so that fields could be edited without requiring the full database metadata ([PR here](https://github.com/metabase/metabase/pull/30484)). When this work was done, we dropped humanizing the coercion strategies, and later on the files that did that string manipulation were removed as they were no longer imported by any files

This PR reverts the commit removing the files, converts them to TS, and adds an e2e test for casting field types

### How to verify
1. Admin -> Table Metadata -> Sample DB -> Feedback
2. Go to the ratings field and click the gear
3. There should be an option to cast the field to a different data type (this is being interpreted as a number, so you're able to cast to UNIX Time)
4. The options in that menu should not be namespaced with `Coercion/`

### Demo
<img width="262" alt="chrome_FZvFE2ygRL" src="https://github.com/user-attachments/assets/40221e24-1c59-47b9-8442-480af38f7dad">

### Checklist
